src/lib.rs

```diff
@@ -48,6 +41,16 @@ macro_rules! seq_t {
     ($ident:ident in u64 $body:tt) => {seq_macro::seq!($ident in 0..64 $body)};
 }
 
+const fn supported_bit_width(width: usize, type_width: usize) -> bool {
+    match type_width {
+        8 => width <= 8,
+        16 => width <= 16,
+        32 => width <= 32,
+        64 => width <= 64,
+        _ => unreachable!(),
+    }
+}
```

Comment:
using const fn and asserting in a `const { }` block has the same effect as the previous trait-based predicates